surface/include/pcl/surface/impl/bilateral_upsampling.hpp

```diff
@@ -116,7 +116,7 @@ pcl::BilateralUpsampling<PointInT, PointOutT>::performProcessing (PointCloudOut
                 abs (input_->points[y_w * input_->width + x_w].g - input_->points[y * input_->width + x].g) +
                 abs (input_->points[y_w * input_->width + x_w].b - input_->points[y * input_->width + x].b));
             
-            float val_exp_rgb = val_exp_rgb_vector(d_color);
+            float val_exp_rgb = val_exp_rgb_vector(int(d_color));
```

Comment:
This needs to be cast to `Eigen::VectorXf::Index` which is the actual index type.